src/sql-parser/src/parser.rs

```diff
@@ -243,7 +441,19 @@ impl<'a> Parser<'a> {
         ParserError { pos, message }
     }
 
-    fn parse_statements(&mut self) -> Result<Vec<Statement<Raw>>, ParserError> {
+    fn statement_error(
+        &self,
+        statement: Option<StatementKind>,
+        pos: usize,
+        message: String,
+    ) -> ParserStatementError {
+        ParserStatementError {
+            error: self.error(pos, message),
+            statement,
+        }
+    }
```

Comment:
This looks like it could also be omitted if we elide `parser_statement_err!` and rely on `parser_err!().map_no_statement_error` everywhere.